hamilton/customization/default.py

```diff
@@ -0,0 +1,303 @@
+        message = f"Executing node: {node_unique_id}."
+        if self.verbosity == 2:
+            message += f" Inputs: \n{pprint.pformat(node_kwargs)}"
+        self.print_fn(message)
+
+    def run_after_node_execution(
+        self,
+        *,
+        node_name: str,
+        node_tags: Dict[str, Any],
+        node_kwargs: Dict[str, Any],
+        result: Any,
+        error: Optional[Exception],
+        success: bool,
+        task_id: Optional[str],
+        **future_kwargs: Any,
+    ):
+        """Runs after a node executes. Prints out the node name and time it took, the output if verbosity is 1.
+
+        :param node_name: Name of the node
+        :param node_tags: Tags of the node
+        :param node_kwargs: Keyword arguments passed to the node
+        :param result: Result of the node
+        :param error: Error of the node
+        :param success: Whether the node was successful or not
+        :param task_id: ID of the task that the node is in, if any
+        :param future_kwargs: Additional keyword arguments that may be passed to the hook yet are ignored for now
+        """
+        if not should_run_node(node_name, node_tags, self.node_filter):
+            return
+        node_unique_id = self._format_node_name(node_name, task_id)
+        time_delta = time.time() - self.timer_dict[node_unique_id]
+        time_delta_formatted = self._format_time_delta(time_delta)
+        message = f"Finished executing node: {node_unique_id} in {time_delta_formatted}. Status: {'Success' if success else 'Failure'}."
+        if self.verbosity == 2:
+            if success:
+                message += f" Result: \n{pprint.pformat(result)}\n"
+            else:
+                message += f" Error: \n{pprint.pformat(error)}"
+        self.print_fn(message)
+        del self.timer_dict[node_name]
+
+
+class PDBDebugger(NodeExecutionHook, NodeExecutionMethod):
+    """Class to inject a PDB debugger into a node execution."""
+
+    CONTEXT = dict()
+
+    def run_to_execute_node(
+        self,
+        *,
+        node_name: str,
+        node_tags: Dict[str, Any],
+        node_callable: Any,
+        node_kwargs: Dict[str, Any],
+        task_id: Optional[str],
+        **future_kwargs: Any,
+    ) -> Any:
+        """Executes the node with a PDB debugger. This modifies the global PDBDebugger.CONTEXT variable to contain information about the node,
+        so you can access it while debugging.
+
+        :param node_name: Name of the node
+        :param node_tags: Tags of the node
+        :param node_callable: Callable function of the node
+        :param node_kwargs: Keyword arguments passed to the node
+        :param task_id: ID of the task that the node is in, if any
+        :param future_kwargs: Additional keyword arguments that may be passed to the hook yet are ignored for now
+        :return: Result of the node
+        """
+        if not should_run_node(node_name, node_tags, self.node_filter):
```

Comment:
there's also a few other things to consider:

1. you can skip stacks from some modules when you pass that as "skip" https://docs.python.org/3/library/pdb.html#pdb.Pdb. could be useful to skip most of hamilton stuff?
2. the post mortem option -- so could try/except around the function and then only debug on error.

Reply:
Yeah skipping stacks is a more powerful way to do this I think? Its a little bit iffy though, so I'm happy to add over time. And yeah, saw the `.pm` option, which should be good for error. We can also add another param `error=True/False`. 

Going to make a comment that the API will likely remain backwards compatible, but the impact of it might not.